linguistics/src/main/java/com/yahoo/language/process/Generator.java

```diff
@@ -0,0 +1,128 @@
+package com.yahoo.language.process;
+
+import com.yahoo.collections.LazyMap;
+import com.yahoo.document.DataType;
+import com.yahoo.language.Language;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+public interface Generator {
+
+    // Name of generator when none is explicitly given
+    String defaultGeneratorId = "default";
+
+    // An instance of this which throws IllegalStateException if attempted used
+    Generator throwsOnUse = new FailingGenerator();
+
+    // Returns this generator instance as a map with the default generator name
+    default Map<String, Generator> asMap() {
+        return asMap(defaultGeneratorId);
+    }
+
+    // Returns this generator instance as a map with the given name
+    default Map<String, Generator> asMap(String name) {
+        return Map.of(name, this);
+    }
+
+    String generate(String prompt, Context context, DataType dataType);
```

Comment:
Please remove the DataType argument, I don't think you need it and sine it is a document side concept it makes it weird to use this interface in other contexts.